dashboard/app/controllers/tracking_pixel_controller.rb

```diff
@@ -0,0 +1,22 @@
+require 'cdo/firehose'
+
+class TrackingPixelController < ApplicationController
+  def index
+    prevent_caching
+    curriculum_page = URI.unescape(params[:from]).split('/').reverse.drop(1).reverse.join('/')
+    user_id = current_user&.id || nil
```

Comment:
The `|| nil` is redundant - the safe-navigation operator evaluates to `nil` if its left-hand side is `nil`.
```suggestion
    user_id = current_user&.id
```